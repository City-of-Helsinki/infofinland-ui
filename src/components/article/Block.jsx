// import { ARTICLE_MARGIN_CLASS } from './Article'
import cls from 'classnames'

import { BLOCK_MARGIN } from '@/components/article/Article'

const Block = ({ children, className }) => (
  <div className={cls(BLOCK_MARGIN, className)}>{children}</div>
)

export default Block
