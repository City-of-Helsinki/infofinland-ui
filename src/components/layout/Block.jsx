import cls from 'classnames'

const Block = ({ children, className, hero, about }) => {

  return (
    <div
      className={cls(className, {
        'ifu-block--article': !hero && !about,
        'ifu-block--hero': hero,
        'ifu-block--secondary': about,
      })}
    >
      {children}
    </div>
  )
}

export default Block
