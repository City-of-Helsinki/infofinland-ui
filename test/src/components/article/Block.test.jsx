/**
 * @jest-environment jsdom
 */

import React from 'react'
// Using render and screen from test-utils.js instead of
// @testing-library/react
// import { render } from "@/test/test-utils";
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react'
import Block from '@/components/article/Block'

describe('Block', () => {
  it('should render the Block', () => {
    const { asFragment } = render(
      <Block>
        <p>Paragraph content</p>
      </Block>
    )
    // we can only use toBeInTheDocument because it was imported
    // in the jest.setup.js and configured in jest.config.js
    expect(asFragment()).toMatchSnapshot()
  })
})
