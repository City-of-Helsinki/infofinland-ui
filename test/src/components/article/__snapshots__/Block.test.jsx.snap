// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block should render the Block 1`] = `
<DocumentFragment>
  <div
    class="ifu-block--article"
  >
    <p>
      Paragraph content
    </p>
  </div>
</DocumentFragment>
`;
